python-backend/app/api/utils/include/user_info.py

```diff
@@ -22,13 +22,13 @@ def get_user_raw_info(self):
 
     def get_user_email(self):
         raw_info = self.get_user_raw_info()
-        return raw_info['email']
+        return raw_info.get('email')
 
     def get_user_given_name(self):
         raw_info = self.get_user_raw_info()
-        return raw_info['given_name']
+        return raw_info.get('given_name')
 
     def get_user_username(self):
         raw_info = self.get_user_raw_info()
         realms = list(set(VALID_REALM) & set(raw_info['realm_access']['roles']))
-        return realms[0] + '\\' + raw_info['preferred_username'] if realms else raw_info['preferred_username']
+        return realms[0] + '\\' + raw_info.get('preferred_username')
```

Comment:
Also this if statement seemed to be ineffective as far I could tell. Whoever . the original author was, please confirm I didn;t misunderstood some complex logic

Reply:
my understanding was this gets processed as A + (B if condition else B) So i changed it to A + B.

If twas doing (A+B if condition else B). Please let me know so i can update that